iceoryx_hoofs/include/iceoryx_hoofs/error_reporting/api.hpp

```diff
@@ -0,0 +1,140 @@
+            else                                                                                                       \
+            {                                                                                                          \
+                iox::err::forwardNonFatalError(iox::err::toError(error), kind, CURRENT_SOURCE_LOCATION);               \
+            }                                                                                                          \
+        }                                                                                                              \
+    } while (false)
+
+/// @brief report fatal error if expr evaluates to false
+/// @note for conditions that may actually happen during correct use
+/// @param expr boolean expression that must hold
+/// @param error error object (or code)
+#define IOX_REQUIRE(expr, error) IOX_REPORT_IF(!(expr), error, FATAL)
+
+//************************************************************************************************
+//* For documentation of intent, defensive programming and debugging
+//*
+//* There are no error codes/errors required here on purpose, as it would make the use cumbersome.
+//* Instead a special internal error type is used.
+//************************************************************************************************
+
+/// @brief if enabled: report fatal precondition violation if expr evaluates to false
+/// @param expr boolean expression that must hold upon entry of the function it appears in
+/// @param message message to be forwarded in case of violation
+#define IOX_PRECONDITION(expr, message)                                                                                \
+    do                                                                                                                 \
+    {                                                                                                                  \
+        if (iox::err::Configuration::CHECK_PRECONDITIONS && !(expr))                                                   \
+        {                                                                                                              \
+            iox::err::forwardFatalError(iox::err::Violation::createPreconditionViolation(),                            \
+                                        iox::err::PRECONDITION_VIOLATION,                                              \
+                                        CURRENT_SOURCE_LOCATION,                                                       \
+                                        message);                                                                      \
+        }                                                                                                              \
+    } while (false)
+
+/// @brief if enabled: report (fatal) assumption violation if expr evaluates to false
```

Comment:
Why is fatal in parenthesis? There is always a fatal error reported.

Reply:
True. It was more that I thought it is not of immediate interest (extra info). Will remove parens.